### DISCUSSION TITLE
EV Charger support

### DISCUSSION THREAD

**Discussion by mikedeluca** [WillCodeForCats/solaredge-modbus-multi#564] [Ideas]
So I was digging around to try to figure out why some sites batteries work and at others they don't show up and I ran across a Modbus address that wasn't one of the 3 inverters. 

{'c_id': 'SunS', 'c_did': 1, 'c_length': 65, 'c_manufacturer': 'SolarEdge', 'c_model': 'SE-EV-SA-US-40P', 'c_version': '0004.0019.0036', 'c_serialnumber': '63002925', 'c_deviceaddress': 4, 'c_sunspec_did': 103, 'c_sunspec_length': 50, 'current': 0, 'l1_current': 0, 'l2_current': 0, 'l3_current': 0, 'current_scale': -32768, 'l1_voltage': 0, 'l2_voltage': 0, 'l3_voltage': 0, 'l1n_voltage': 0, 'l2n_voltage': 0, 'l3n_voltage': 0, 'voltage_scale': -32768, 'power_ac': -32768, 'power_ac_scale': -32768, 'frequency': 0, 'frequency_scale': -32768, 'power_apparent': -32768, 'power_apparent_scale': -32768, 'power_reactive': -32768, 'power_reactive_scale': -32768, 'power_factor': -32768, 'power_factor_scale': -32768, 'energy_total': 0, 'energy_total_scale': -32768, 'current_dc': 0, 'current_dc_scale': -32768, 'voltage_dc': 0, 'voltage_dc_scale': -32768, 'power_dc': -32768, 'power_dc_scale': -32768, 'temperature': -32768, 'temperature_scale': -32768, 'status': 1, 'vendor_status': 0, 'rrcr_state': 0, 'active_power_limit': 100, 'cosphi': 0, 'commit_power_control_settings': 0, 'restore_power_control_default_settings': 0, 'reactive_power_config': 0, 'reactive_power_response_time': 0, 'advanced_power_control_enable': 0, 'export_control_mode': 0, 'export_control_limit_mode': 0, 'export_control_site_limit': 0}

This is ID 4 on the Modbus TCP. The model and serial match my Solaredge EV Charger. Feels like this would be interesting functionality to add... I'll try to do some more scanning of registers

**Comment by WillCodeForCats** [comment]
Sure, I can do that. Thank for the details.

**Comment by WillCodeForCats** [comment]
The first set of registers appears to follow the sunspec identifying block (`c_id` through `c_sunspec_length`), but after that it's not an inverter so I assume those are going to be unique to the car charger. For example the `c_sunspec_did` value of `103` is for a three phase inverter, other than the model name there doesn't appear to be anything that uniquely identifies the EV charger.

Since there isn't any modbus info I'm aware of for the car charger you will need to try different state combinations like charging, unplugged, plugged in but finished charging, and see if any of the register values change.

**Reply by mikedeluca** [reply]
Will do. In the integration I've added it as a 4th inverter for now. Next time I'm plugged in and charging I'll see what it says.

**Reply by will36** [reply]
@mikedeluca did any values come through when you charged?

**Comment by WillCodeForCats** [comment]
I created issue #565 to work on this.
